extensions/oidc/runtime/src/main/java/io/quarkus/oidc/VertxKeycloakRecorder.java

```diff
@@ -62,4 +62,22 @@ public void handle(AsyncResult<OAuth2Auth> event) {
         mechanism.setAuth(auth);
         mechanism.setAuthServerURI(config.authServerUrl);
     }
+
+    protected static OIDCException toOidcException(Throwable cause) {
+        final String message = "OIDC server is not available at the 'quarkus.oidc.auth-server-url' URL. "
+                + "Please make sure it is correct. Note it has to end with a realm value if you work with Keycloak";
+        Throwable vertxCause = findVertxCause(cause);
+        return new OIDCException(message, vertxCause);
+    }
+
+    private static Throwable findVertxCause(Throwable cause) {
+        if (cause == null) {
+            return null;
+        }
+        if (cause.getClass().getName().contains("vertx")) {
+            return cause;
+        } else {
+            return findVertxCause(cause);
```

Comment:
Is it me or this will lead to an infinite loop?

Reply:
Oops, it will. 1 sec

Reply:
I don't recall seeing a stack overflow message though, but I guess it only because the 1st cause was the vertx one